### DISCUSSION TITLE
Verification of phone numbers?

### DISCUSSION THREAD

**Discussion by landerss1** [ory/kratos#2555] [Q&A] [answered]
Hi,

We have a need to create and store a type of identity which have an email, a phone number, or both, nothing else as traits. As I understand, there has been some work done in the area of using phone as a way to perform recovery and verification. There has also been some work in the courier to be able to send sms using the phone numbers. I understand the configuration that needs to be done and the identity schema, but now I'm looking at the verification and recovery flows and there's nothing there that mentions how to use the phone number in these flows. Am I missing something or is it currently not possible to verify a phone number via sms?

**Comment by kmherrmann** [comment, marked as answer]
Hi there - you're right, SMS verification isn't yet available. However, it's on our roadmap, and there's also work going on [in this PR](https://github.com/ory/kratos/pull/2511) that might be helpful.

**Reply by landerss1** [reply]
@kmherman Thanks. I used the PR and filled in the gaps to create a temporary solution using our own fork. Unfortunately, due to summer vacation and other issues there's not enough time for me to contribute anything this time. If you go forward with the implementation, may I suggest to use the current flow for email verification and only modify the API so it can accept **either** `phone `or `email `in the request body of the complete verification flow? That way the change to the API is minimal:

`curl -X POST "http://<host>:4433/self-service/verification?flow=<flow-id>" -H "Content-Type: application/json" -d "{\"phone\":\"+461234567\",\"method\":\"link\"}"`

**Reply by vinckr** [reply]
Hello @avamonitoring 
Can you add this comment to the PR as well please 🙏

**Reply by landerss1** [reply]
Absolutely! Just did it.
